### ISSUE TITLE
Remove references to React lifecycle methods deprecated in 16.3

### ISSUE THREAD

**Issue by claudiopro** [facebook/draft-js#1940]
React components in Draft.js still use or reference lifecycle methods that were deprecated in React 16.3 because if misused, they can get in the way of asynchronous rendering.

The deprecated methods are:
```
componentWillMount
componentWillReceiveProps
componentWillUpdate
```

[This post on the React blog](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html#migrating-from-legacy-lifecycles) provides an excellent migration guide and offers common scenarios with recipes to replace the legacy lifecycle methods with the new API.

It seems we only have references in examples or in code comments, so this makes it an easy first bug!

**Comment by danedavid** [comment]
Can I take this one up?

**Comment by claudiopro** [comment]
Hi @danedavid, thanks for taking a look! I am afraid this issue is closed already with #1952. Please check back again for issues tagged with "good first issue".
